### PR TITLE
Hardware Firewall (Dedicated) and rules

### PR DESCRIPTION
**Pull request by minsikl**
Define `softlayer_fw_hardware_dedicated` for firewall device management. 
Define`softlayer_fw_hardware_dedicated_rules` to configure and manage firewall rules.

### CODE REVIEW

File: docs/resources/softlayer_fw_hardware_dedicated_rules.md [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+[Configure a Hardware Firewall (Dedicated)](https://knowledgelayer.softlayer.com/procedure/configure-hardware-firewall-dedicated).
+
+_Please Note_: Target VLAN should have at least one subnet for rule configuration.
+
+_Please Note_: To express _ANY IP addresses_ in external side, configure `src_ip_address` as `0.0.0.0` and `src_ip_cidr` as `0`.
+
+_Please Note_: To express _API IP addresses_ in internal side, configure `dst_ip_address` as `any` and `src_ip_cidr` as `32`.
+
+_Please Note_: Firewall rule removal is not allowed in SoftLayer. If terraform destroys `softlayer_fw_hardware_dedicated_rules` resources, 
+_permit from any to any with TCP, UDP, ICMP, GRE, PPTP, ESP, and HA_ rules will be configured.
```
Comment (renier) [inline docs/resources/softlayer_fw_hardware_dedicated_rules.md]: I would put all these notes under the same paragraph.

File: softlayer/resource_softlayer_fw_hardware_dedicated.go [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+
+	haEnabled := d.Get("ha_enabled").(bool)
+	publicVlanId := d.Get("public_vlan_id").(int)
+
+	var keyName string
+	if haEnabled {
+		keyName = "HARDWARE_FIREWALL_HIGH_AVAILABILITY"
+	} else {
+		keyName = "HARDWARE_FIREWALL_DEDICATED"
+	}
```
Comment (renier) [inline softlayer/resource_softlayer_fw_hardware_dedicated.go:60]:
```go
// Set the default first, then switch to ha if set
keyName := "HARDWARE_FIREWALL_DEDICATED"
if haEnabled {
    keyName = "HARDWARE_FIREWALL_HIGH_AVAILABILITY"
}
```

**Comment by renier** [comment]
Looks good to me. Above are the only comments.

File: docs/resources/softlayer_fw_hardware_dedicated_rules.md (hunk 1 above)
Comment (minsikl) [reply docs/resources/softlayer_fw_hardware_dedicated_rules.md]: Done.

File: softlayer/resource_softlayer_fw_hardware_dedicated.go (hunk 2 above)
Comment (minsikl) [reply softlayer/resource_softlayer_fw_hardware_dedicated.go:60]: Done.

**Pull request merged**